File: 03-array/08-remove-dups-from-sorted-array/08-remove-dups-from-sorted-array.rb
def remove_dups_from_sorted_array(nums)
  n, j = nums.length, 1
  (1..n-1).each do |i|
    if nums[j-1] != nums[i]
      nums[j] = nums[i]
      j += 1
    end
  end
  nums
end

print remove_dups_from_sorted_array [10,20,20,30,30,30,30]

File: 03-array/08-remove-dups-from-sorted-array/08-remove-dups-from-sorted-array.ts
const remove_dups_from_sorted_array = (nums: number[]): number[] => {
  let j = 1
  for (let i = 1; i < nums.length; i++) {
    if (nums[i] != nums[j - 1]) {
      nums[j] = nums[i]
      j += 1
    }
  }
  return nums
}

console.log(remove_dups_from_sorted_array([10, 20, 20, 30, 30, 30, 30]))
